lib/plugins/aws/package/compile/events/cognitoUserPool/index.js

```diff
@@ -32,9 +32,15 @@ class AwsCompileCognitoUserPoolEvents {
       'after:package:finalize': this.mergeWithCustomResources.bind(this),
     };
 
-    // TODO: Complete schema, see https://github.com/serverless/serverless/issues/8028
     this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'cognitoUserPool', {
       type: 'object',
+      properties: {
+        pool: { type: 'string', maxLength: 128, regexp: /^[\w\s+=,.@-]+$/.toString() },
```

Comment:
Is `maxLength` respected together with `regexp`?

Also if we do not need case _insensitivity_ I would rather use `pattern` (`regex` is a suplemential AJV extension, not a JSON schema standard)

Reply:
`maxLength` and `regexp` do work together nicely :)

Updated with `pattern` instead of `regexp`